test/time_test.py

```diff
@@ -19,78 +31,106 @@ def test_get_fps(self):
         c = Clock()
         self.assertEqual(c.get_fps(), 0)
         # Type check get_fps should return float
-        self.assertTrue(type(c.get_fps()) == float)
-        # Allowable margin of error in percentage
-        delta = 0.30
-        # Test fps correctness for 100, 60 and 30 fps
-        self._fps_test(c, 100, delta)
-        self._fps_test(c, 60, delta)
-        self._fps_test(c, 30, delta)
-
-    def _fps_test(self, clock, fps, delta):
+        self.assertTrue(isinstance(c.get_fps(), float))
+
+        # Test fps correctness for a few fps
+        for i in [24, 30, 60, 100, 120]:
+            self._fps_test(c, i)
+
+    def _fps_test(self, clock, fps, reccnt=0):
         """ticks fps times each second, hence get_fps() should return fps"""
-        delay_per_frame = 1.0/fps
-        for f in range(fps):  # For one second tick and sleep
+        if reccnt == NUM_TRIES:
+            self.fail("get_fps does not work")
+
+        delta = 0.2
+        delay_per_frame = 1.0 / fps
+        for f in range(fps // 4):  # For 250ms tick and sleep
             clock.tick()
             time.sleep(delay_per_frame)
+
         # We should get around fps (+- fps*delta -- delta % of fps)
-        self.assertAlmostEqual(clock.get_fps(), fps, delta=fps*delta)
+        if not _is_almost_equal(clock.get_fps(), fps, delta):
+            # recurse on failure
+            self._fps_test(clock, fps, reccnt + 1)
 
     def test_get_rawtime(self):
-
-        iterations = 10
-        delay = 0.1
-        delay_miliseconds = delay*(10**3) #actual time difference between ticks
-        framerate_limit = 5
-        delta = 50 #allowable error in milliseconds
-
-        #Testing Clock Initialization
+        iterations = 5
+        delay = 0.02
+        delay_milliseconds = delay * 1000
+        framerate_limit = 20
+        delta = 0.1
+        errcnt = 0
+
+        # Testing Clock Initialization
         c = Clock()
         self.assertEqual(c.get_rawtime(), 0)
 
-        #Testing Raw Time with Frame Delay
+        # Testing Raw Time with Frame Delay
         for f in range(iterations):
             time.sleep(delay)
             c.tick(framerate_limit)
             c1 = c.get_rawtime()
-            self.assertAlmostEqual(delay_miliseconds, c1, delta=delta)
+            if not _is_almost_equal(delay_milliseconds, c1, delta):
+                errcnt += 1
+
+        if errcnt >= NUM_TRIES:
+            self.fail("error while doing get_rawtime()")
 
-       #Testing get_rawtime() = get_time()
+        errcnt = 0
+
+        # Testing get_rawtime() = get_time()
         for f in range(iterations):
             time.sleep(delay)
             c.tick()
             c1 = c.get_rawtime()
             c2 = c.get_time()
-            self.assertAlmostEqual(c1, c2, delta=delta)
 
+            self.assertEqual(c1, c2)
+            if not _is_almost_equal(delay_miliseconds, c1, delta):
```

Comment:
One remaining typo here:

`delay_miliseconds` here should be `delay_milliseconds` with two l's.

Reply:
Nice spot :)
fixed